app/src/main/java/com/example/paritosh/stormy/Presenter.java

```diff
@@ -0,0 +1,41 @@
+package com.example.paritosh.stormy;
+
+import android.content.Context;
+import android.widget.Toast;
+
+import com.example.paritosh.stormy.databinding.ActivityMainBinding;
+import com.example.paritosh.stormy.model.CurrentWeather;
+import com.example.paritosh.stormy.model.CurrentWeatherDataBindingModel;
+
+public class Presenter {
+
+
+    private WeatherDataProvider weatherDataProvider = new WeatherDataProvider();
+
+    public void updateWeatherDetails(final ActivityMainBinding binding, Context context) {
```

Comment:
The presenter shouldn't hold any information related to libraries used by the view to display data. Don't pass binding and context here. To retrieve the instance of MainActivity pass the instance through the constructor of the presenter and store it in the presenter.